ios/PriceWidget/PriceWidgetView.swift

```diff
@@ -0,0 +1,183 @@
+//
+//  PriceWidgetView.swift
+//  Rainbow
+//
+//  Created by Ben Goldberg on 10/27/21.
+//
+
+import Foundation
+import SwiftUI
+import UIKit
+
+@available(iOS 14.0, *)
+struct PriceWidgetView: View {
+
+  let tokenData: TokenData
+  let date: Date
+  
+  private func toString(double: Double) -> String {
+    return String(format: "%f", double)
+  }
+  
+  private func getIcon(icon: UIImage) -> UIImage {
+    return tokenData.icon!.resizeImageTo(size: CGSize(width: 20, height: 20))
+  }
+  
+  private func convertToCurrency(double: Double) -> String {
+    let nf = NumberFormatter()
+    nf.locale = Locale(identifier: "en_US")
+    nf.usesGroupingSeparator = true
```

Comment:
what do we do with non-USD as a def currency?

Reply:
nothing right now. once i'm able to fetch user preferences, i can modify the TokenData struct to include currency and create a switch statement in the convertToCurrency function to define different formatting styles (i.e. currency symbol and comma/period placement). The CoinGecko api response also contains a bunch of different currencies besides usd that i can choose from for current toke price